src/integration_tests/merge_subscription.rs

```diff
@@ -0,0 +1,223 @@
+use serde_json::json;
+
+use crate::client::{
+    mock::{run_sink_tasks, SinkTask, TestServerBuilder},
+    Client,
+};
+use crate::config::{Config, MergeStrategy, RpcDefinitions, RpcSubscription, ServerConfig};
+use crate::server::start_server;
+
+#[tokio::test]
+async fn merge_subscription_works() {
+    let subscribe_head = "chain_subscribeNewHeads";
+    let update_head = "chain_newHead";
+    let unsubscribe_head = "chain_unsubscribeNewHeads";
+
+    let subscribe_finalized = "chain_subscribeFinalizedHeads";
+    let update_finalized = "chain_finalizedHead";
+    let unsubscribe_finalized = "chain_unsubscribeFinalizedHeads";
+
+    let subscribe_mock = "mock_sub";
+    let unsubscribe_mock = "mock_unsub";
+    let update_mock = "mock";
+
+    let mut builder = TestServerBuilder::new();
+
+    let mut head_sub = builder.register_subscription(subscribe_head, update_head, unsubscribe_head);
+    let mut finalized_sub =
+        builder.register_subscription(subscribe_finalized, update_finalized, unsubscribe_finalized);
+    let mut mock_sub_rx =
+        builder.register_subscription(subscribe_mock, update_mock, unsubscribe_mock);
+
+    let (addr, _upstream_handle) = builder.build().await;
+
+    tokio::spawn(async move {
+        let (_, head_sub_rx) = head_sub.recv().await.unwrap();
+        let (_, finalized_sub_rx) = finalized_sub.recv().await.unwrap();
+
+        run_sink_tasks(&head_sub_rx, vec![SinkTask::Send(json!(1))]).await;
+        run_sink_tasks(&finalized_sub_rx, vec![SinkTask::Send(json!(1))]).await;
+    });
+
+    let config = Config {
+        endpoints: vec![format!("ws://{addr}")],
+        stale_timeout_seconds: 0,
+        server: ServerConfig {
+            listen_address: "0.0.0.0".to_string(),
+            port: 8888,
```

Comment:
```suggestion
            port: 0,
```

always specify port `0` for tests